services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/DefaultS3TransferManager.java

```diff
@@ -402,16 +404,58 @@ private static boolean isMrapArn(Arn arn) {
     }
 
     private static final class DefaultBuilder implements S3TransferManager.Builder {
-        private S3ClientConfiguration s3ClientConfiguration = S3ClientConfiguration.builder().build();
+        private S3AsyncClient s3AsyncClient;
         private S3TransferManagerOverrideConfiguration transferManagerConfiguration =
             S3TransferManagerOverrideConfiguration.builder().build();
 
         private DefaultBuilder() {
         }
 
+        /**
+         * If S3AsyncClient is not supplied then SDK wil create default S3AsyncClient with following properties
+         *
+         * <p>
+         *     If region is not specified, the SDK will attempt to identify the endpoint automatically using the following logic:
+         * <ol>
+         *     <li>Check the 'aws.region' system property for the region.</li>
+         *     <li>Check the 'AWS_REGION' environment variable for the region.</li>
+         *     <li>Check the {user.home}/.aws/credentials and {user.home}/.aws/config files for the region.</li>
+         *     <li>If running in EC2, check the EC2 metadata service for the region.</li>
+         * </ol>
+         *
+         * <p>The default provider will attempt to identify the credentials automatically using the following checks:
+         * <ol>
+         *   <li>Java System Properties - <code>aws.accessKeyId</code> and <code>aws.secretKey</code></li>
+         *   <li>Environment Variables - <code>AWS_ACCESS_KEY_ID</code> and <code>AWS_SECRET_ACCESS_KEY</code></li>
+         *   <li>Credential profiles file at the default location (~/.aws/credentials) shared by all AWS SDKs and the AWS CLI</li>
+         *   <li>Credentials delivered through the Amazon EC2 container service if AWS_CONTAINER_CREDENTIALS_RELATIVE_URI
+         *   environment variable is set and security manager has permission to access the variable.</li>
+         *   <li>Instance profile credentials delivered through the Amazon EC2 metadata service</li>
+         * </ol>
+         *
+         * <p>If the credentials are not found in any of the locations above, an exception will be thrown at {@link #build()}
+         * time.
+         * </p>
+         * <p>
+         *     If Minimum part size is not specified then by default, it is 8MB
+         * </p>
+         *
+         * <p>
+         *     If target throughput for transfer requests provided, the TransferManager
+         *     will calculate the optional number of connections based on {@link #targetThroughputInGbps}.
+         *     If the value is too low, the S3TransferManager might not achieve the specified target throughput.
+         * </p>
+         *
+         * <p>
+         *     If maximum number of S3 connections
+         * </p>
```

Comment:
Sorry for being unclear, I meant adding the Javadocs to the corresponding methods in the `S3CrtAsyncClientBuilder `class

Reply:
Added now